modules/auxiliary/scanner/scada/modbus_banner_grabbing.rb

```diff
@@ -167,21 +167,21 @@ def run_host(ip)
       end
     rescue ::Interrupt
       print_error('MODBUS - Interrupt during payload')
-      raise $!
+      raise $ERROR_INFO
     rescue ::Rex::HostUnreachable, ::Rex::ConnectionError, ::Rex::ConnectionTimeout, ::Rex::ConnectionRefused => e
       print_error("MODBUS - Network error during payload: #{e}")
       return
     rescue ::EOFError
       print_error('MODBUS - No reply')
       return
     end
+  end
 
-    def cleanup
```

Comment:
This was actually a previously nested method. Oops.

Reply:
Before:

The cleanup method was _inside_ `run_host`:

```ruby
def run_host(...)
   def cleanup
      ...
   end
end
```

After:

`cleanup` is now at the right level and performs the correct cleanup:

```ruby
def run_host(...)
   ...
end

def cleanup
   ...
end
```

Reply:
Thanks for explaining this, previous comment makes much more sense now!